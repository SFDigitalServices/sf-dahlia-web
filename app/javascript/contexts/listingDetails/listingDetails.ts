interface ListingDetails {
  /**  */
  uid: string

  /**  */
  email: string

  /**  */
  created_at: Date

  /**  */
  updated_at: Date
}

interface ListingDetailsData {
  data: any
  success: boolean
}

export { ListingDetails, ListingDetailsData }
