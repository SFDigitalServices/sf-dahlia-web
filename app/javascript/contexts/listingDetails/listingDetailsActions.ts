import { createAction } from "typesafe-actions"
import { RailsUnit } from "../../api/types/rails/listings/RailsUnit"
import { RailsAmiChart } from "../../api/types/rails/listings/RailsAmiChart"

export enum ListingDetailsActions {
  StartFetchingUnits = "SetFetchingUnits",
  FinishFetchingUnits = "FinishFetchingUnits",
  SetUnits = "SetUnits",
  SetAmiCharts = "SetAmiCharts",
  StartFetchingAmiCharts = "SetFetchingAmiCharts",
  SetFetchedAmiCharts = "SetFetchedAmiCharts",
  SetFetchingAmiChartsError = "SetFetchingAmiChartsError",
  SetFetchingUnitsError = "SetFetchingUnitsError",
  FinishFetchingAmiCharts = "FinishFetchingAmiCharts",
}

export const startFetchingUnits = createAction(ListingDetailsActions.StartFetchingUnits)()
export const setUnits = createAction(ListingDetailsActions.SetUnits)()
export const setAmiCharts = createAction(ListingDetailsActions.SetAmiCharts)()
export const finishFetchingAmiCharts = createAction(ListingDetailsActions.FinishFetchingAmiCharts)<
  RailsAmiChart[]
>()
export const setFetchedAmiCharts = createAction(ListingDetailsActions.SetFetchedAmiCharts)()
export const startFetchingAmiCharts = createAction(ListingDetailsActions.StartFetchingAmiCharts)()
export const finishFetchingUnits = createAction(ListingDetailsActions.FinishFetchingUnits)<
  RailsUnit[]
>()
export const setFetchingUnitsError = createAction(
  ListingDetailsActions.SetFetchingUnitsError
)<Error>()
export const setFetchingAmiChartsError = createAction(
  ListingDetailsActions.SetFetchingAmiChartsError
)<Error>()
