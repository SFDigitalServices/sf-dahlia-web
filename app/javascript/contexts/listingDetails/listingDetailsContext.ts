import { createContext } from "react"
import RailsUnit from "../../api/types/rails/listings/RailsUnit"
import { RailsAmiChart, RailsAmiChartMetaData } from "../../api/types/rails/listings/RailsAmiChart"

export type ContextProps = {
  units: RailsUnit[]
  amiCharts: RailsAmiChart[]
  fetchedAmiCharts: boolean
  fetchingAmiCharts: boolean
  fetchedUnits: boolean
  fetchingUnits: boolean
  fetchUnits: (listingId: string) => void
  fetchAmiCharts: (chartsToFetch: RailsAmiChartMetaData[]) => void
  fetchingAmiChartsError: Error
  fetchingUnitsError: Error
}

export default createContext<Partial<ContextProps>>({})
