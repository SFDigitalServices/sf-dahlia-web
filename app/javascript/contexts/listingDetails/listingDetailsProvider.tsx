import React, { useReducer } from "react"
import ListingDetailsReducer from "./listingDetailsReducer"
import ListingDetailsContext, { ContextProps } from "./listingDetailsContext"
import { getUnits, getAmiCharts } from "../../api/listingApiService"
import { RailsUnit } from "../../api/types/rails/listings/RailsUnit"
import { RailsAmiChart, RailsAmiChartMetaData } from "../../api/types/rails/listings/RailsAmiChart"
import {
  finishFetchingUnits,
  finishFetchingAmiCharts,
  startFetchingAmiCharts,
  startFetchingUnits,
  setFetchingAmiChartsError,
  setFetchingUnitsError,
} from "./listingDetailsActions"

interface ListingDetailsProviderProps {
  children?: React.ReactNode
}

const ListingDetailsProvider = (props: ListingDetailsProviderProps) => {
  const [state, dispatch] = useReducer(ListingDetailsReducer, {
    units: [],
    amiCharts: [],
    fetchingUnits: false,
    fetchedUnits: false,
    fetchingAmiCharts: false,
    fetchedAmiCharts: false,
    fetchingAmiChartsError: null,
    fetchingUnitsError: null,
  })

  const contextValues: ContextProps = {
    units: state.units,
    amiCharts: state.amiCharts,
    fetchedAmiCharts: state.fetchedAmiCharts,
    fetchingAmiCharts: state.fetchingAmiCharts,
    fetchedUnits: state.fetchedUnits,
    fetchingUnits: state.fetchingUnits,
    fetchingAmiChartsError: state.fetchingAmiChartsError,
    fetchingUnitsError: state.fetchingUnitsError,
    fetchUnits: (listingId: string) => {
      dispatch(startFetchingUnits())
      getUnits(listingId)
        .then((units: RailsUnit[]) => {
          dispatch(finishFetchingUnits(units))
        })
        .catch((error) => {
          console.error(error)
          dispatch(setFetchingUnitsError(error))
        })
    },
    fetchAmiCharts: (chartsToFetch: RailsAmiChartMetaData[]) => {
      dispatch(startFetchingAmiCharts())
      getAmiCharts(chartsToFetch)
        .then((amiCharts: RailsAmiChart[]) => {
          dispatch(finishFetchingAmiCharts(amiCharts))
        })
        .catch((error) => {
          console.error(error)
          dispatch(setFetchingAmiChartsError(error))
        })
    },
  }

  return (
    <ListingDetailsContext.Provider value={contextValues}>
      {props.children}
    </ListingDetailsContext.Provider>
  )
}

export default ListingDetailsProvider
