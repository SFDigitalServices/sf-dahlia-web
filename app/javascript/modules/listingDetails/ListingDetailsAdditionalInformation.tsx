import React from "react"
import { LinkButton, ListingDetailItem, t } from "@bloom-housing/ui-components"
import { RailsListing } from "../listings/SharedHelpers"
import { TextTruncate } from "../../components/TextTruncate"
import { isHabitatListing, isSale } from "../../util/listingUtil"
import { stripMostTags } from "../../util/filterUtil"
import { getTranslatedString } from "../../util/languageUtil"

export interface ListingDetailsAdditionalInformationProps {
  listing: RailsListing
  imageSrc: string
}

export const ListingDetailsAdditionalInformation = ({
  listing,
  imageSrc,
}: ListingDetailsAdditionalInformationProps) => {
  const getCommissionString = () => {
    return listing.Realtor_Commission_Unit === "percent"
      ? t("listings.realtorCommissionPercentage", {
          percentage: listing.Realtor_Commission_Amount,
        })
      : `$${listing.Realtor_Commission_Amount.toLocaleString()}`
  }

  return (
    <ListingDetailItem
      imageAlt={""}
      imageSrc={imageSrc}
      title={t("listings.additionalInformation.header")}
      subtitle={t("listings.additionalInformation.subheader")}
    >
      <div className="listing-detail-panel">
        {listing.Listing_Other_Notes && (
          <div className="info-card bg-gray-100 border-0">
            <h3 className="text-serif-xl">{t("listings.specialNotes")}</h3>
            <TextTruncate
              className="primary-lighter-markup-link translate"
              buttonClassName="text-blue-700"
              text={stripMostTags(
                getTranslatedString(
                  listing.Listing_Other_Notes,
                  "Listing_Other_Notes",
                  listing.translations
                )
              )}
            />
          </div>
        )}
        {(!!listing.Required_Documents || isSale(listing)) && (
          <div className="info-card bg-gray-100 border-0">
            <h3 className="text-serif-xl">{t("listings.requiredDocuments")}</h3>
            <div className="text-xs">
              <TextTruncate
                className="primary-lighter-markup-link translate"
                buttonClassName="text-blue-700"
                text={stripMostTags(
                  getTranslatedString(
                    listing.Required_Documents,
                    "Required_Documents",
                    listing.translations
                  )
                )}
              />
            </div>
            {isSale(listing) && !isHabitatListing(listing) && (
              <div className="text-xs mt-4">
                <TextTruncate
                  className="primary-lighter-markup-link translate"
                  buttonClassName="text-blue-700"
                  text={t("listings.requiredDocumentsAfterApplying", {
                    url: "https://sfmohcd.org/after-homebuyer-lottery",
                  })}
                />
              </div>
            )}
          </div>
        )}
        {listing.Legal_Disclaimers && (
          <div className="info-card bg-gray-100 border-0">
            <h3 className="text-serif-xl">{t("listings.importantProgramRules")}</h3>
            <div className="text-xs">
              <TextTruncate
                text={stripMostTags(
                  getTranslatedString(
                    listing.Legal_Disclaimers,
                    "Legal_Disclaimers",
                    listing.translations
                  )
                )}
                className="primary-lighter-markup-link translate"
                buttonClassName="text-blue-500"
              />
            </div>
          </div>
        )}
        {listing.CC_and_R_URL && (
          <div className="info-card bg-gray-100 border-0">
            <h3 className="text-serif-xl">{t("listings.cc&r")}</h3>
            <div className="text-xs">
              <TextTruncate
                className="primary-lighter-markup-link translate"
                buttonClassName="text-blue-500"
                text={t("listings.cc&rDescription")}
              />
              <br />
              <LinkButton href={listing.CC_and_R_URL} className={"mt-4"} newTab={true}>
                {t("listings.downloadPdf")}
              </LinkButton>
            </div>
          </div>
        )}
        {isSale(listing) && (
          <div className="info-card bg-gray-100 border-0">
            <h3 className="text-serif-xl">{t("listings.realtorCommission")}</h3>
            <div className="text-xs">
              {listing.Allows_Realtor_Commission ? (
                <>
                  <span className={"font-bold"}>{t("listings.realtorCommissionHeader")}</span>
                  {getCommissionString()}
                </>
              ) : (
                t("listings.realtorCommissionNotEligible")
              )}
              {listing.Realtor_Commission_Info && (
                <div className={"mt-4"}>
                  <span className={"font-bold"}>{t("listings.realtorCommissionHowTo")}</span>
                  <span>
                    <TextTruncate
                      className="primary-lighter-markup-link translate"
                      buttonClassName="text-blue-500"
                      text={stripMostTags(
                        getTranslatedString(
                          listing.Realtor_Commission_Info,
                          "Realtor_Commission_Info",
                          listing.translations
                        )
                      )}
                    />
                  </span>
                </div>
              )}
            </div>
          </div>
        )}
        {/* TODO: implement once we've established needs for sales listings
              {listing.isSale && (
                <div className="info-card bg-gray-100 border-0">
                  <h3 className="text-serif-xl">For the Buyer's Realtor</h3>
                  {listing.Allows_Realtor_Commission ? (
                    display realtor_commission_header
                    realtorComissionMessage
                    {listing.Realtor_Commission_Info && realtor_commission_how_to}
                  ) : display realtor_commission_not_eligible message}
                </div>
              )}
            */}
        {listing.Repricing_Mechanism && (
          <div className="info-card bg-gray-100 border-0">
            <h3 className="text-serif-xl">{t("listings.rePricing")}</h3>
            <div className="text-xs">
              <TextTruncate
                className="primary-lighter-markup-link translate"
                buttonClassName="text-blue-500"
                text={stripMostTags(
                  getTranslatedString(
                    listing.Repricing_Mechanism,
                    "Repricing_Mechanism",
                    listing.translations
                  )
                )}
              />
            </div>
          </div>
        )}
      </div>
    </ListingDetailItem>
  )
}
