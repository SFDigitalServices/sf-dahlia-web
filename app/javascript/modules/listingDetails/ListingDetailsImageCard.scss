.share-button {
  &:focus {
    @apply rounded-full;
    transition: all 200ms ease-out;
    outline: none;
    box-shadow: 0 0 0 2px #fff, 0 0 3px 4px var(--bloom-color-accent-cool);
  }
}
