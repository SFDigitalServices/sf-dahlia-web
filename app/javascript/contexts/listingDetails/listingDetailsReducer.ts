import { createReducer } from "typesafe-actions"
import { ListingDetailsActions } from "./listingDetailsActions"
import { RailsUnit } from "../../api/types/rails/listings/RailsUnit"
import { RailsAmiChart } from "../../api/types/rails/listings/RailsAmiChart"
import { AmiChartData } from "./listingDetailsContext"

type ListingDetailsState = {
  fetchingUnits: boolean
  fetchedUnits: boolean
  fetchingAmiCharts: boolean
  fetchedAmiCharts: boolean
  units: RailsUnit[]
  amiChartData: AmiChartData
  amiCharts: RailsAmiChart[]
  fetchingAmiChartsError: Error
  fetchingUnitsError: Error
}

const ListingDetailsReducer = createReducer({ units: [] } as ListingDetailsState, {
  [ListingDetailsActions.StartFetchingUnits]: (state) => {
    return { ...state, fetchingUnits: true }
  },
  [ListingDetailsActions.StartFetchingAmiCharts]: (state) => {
    return { ...state, fetchingAmiCharts: true }
  },
  [ListingDetailsActions.FinishFetchingUnits]: (state, { payload: units }) => {
    const amiChartYears: Array<number> = []
    const amiChartTypes = []
    const amiChartPercentages = []

    units.forEach((unit: RailsUnit) => {
      const amiChartYear = amiChartYears.find((year) => {
        return year === unit.AMI_chart_year
      })

      const amiChartType = amiChartTypes.find((type) => {
        return type === unit.AMI_chart_type
      })

      const amiChartPercentage = amiChartPercentages.find((type) => {
        return type === unit.Max_AMI_for_Qualifying_Unit
      })

      if (!amiChartYear) {
        amiChartYears.push(unit.AMI_chart_year)
      }

      if (!amiChartType) {
        amiChartTypes.push(unit.AMI_chart_type)
      }

      if (!amiChartPercentage) {
        amiChartPercentages.push(unit.Max_AMI_for_Qualifying_Unit)
      }
    })

    return {
      ...state,
      fetchingUnits: false,
      fetchedUnits: true,
      units,
      amiChartData: {
        years: amiChartYears,
        percentages: amiChartPercentages,
        types: amiChartTypes,
      },
    }
  },
  [ListingDetailsActions.FinishFetchingAmiCharts]: (state, { payload: amiCharts }) => {
    return {
      ...state,
      fetchingAmiCharts: false,
      fetchedAmiCharts: true,
      amiCharts: amiCharts?.map((amiChart: RailsAmiChart) => {
        return {
          ...amiChart,
          chartType: amiChart?.values[0]?.chartType,
        }
      }),
    }
  },
})

export default ListingDetailsReducer
