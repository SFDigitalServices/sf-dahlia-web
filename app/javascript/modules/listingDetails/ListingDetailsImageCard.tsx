import React, { useContext, useState } from "react"
import { ImageCard, t } from "@bloom-housing/ui-components"
import { getReservedCommunityType } from "../../util/languageUtil"
import type { ImageItem } from "@bloom-housing/ui-components"
import { RailsListing } from "../listings/SharedHelpers"
import { getShareListingPath } from "../../util/routeUtil"
import { getListingAddressString } from "../../util/listingUtil"
import { ConfigContext } from "../../lib/ConfigContext"
import { ListingAddress } from "../../components/ListingAddress"
import fallbackImg from "../../../assets/images/bg@1200.jpg"
import "./ListingDetailsImageCard.scss"

export interface ListingDetailsImageCardProps {
  listing: RailsListing
}

const createImageCardProps = (listing: RailsListing) => {
  const listingImages: ImageItem[] = listing?.Listing_Images?.map((listingImage) => {
    return {
      url: listingImage.Image_URL,
      description: listingImage.Image_Description,
    }
  })

  // We want to support both the imageURL and listing images fields for now
  // but the hope is to ultimately deprecate that field.
  // If we have to use the fallback image then we want to disable the image
  // field from A11Y tools since at that point it is purely decorative.
  if (!listingImages) {
    return listing?.imageURL
      ? {
          props: {
            imageUrl: listing.imageURL,
            description: t("listings.buildingImageAltText"),
          },
          fallbackUsed: false,
        }
      : {
          props: {
            imageUrl: fallbackImg,
            description: "",
          },
          fallbackUsed: true,
        }
  } else {
    return listingImages.length > 0
      ? {
          props: {
            images: listingImages,
            description: t("listings.buildingImageAltText"),
            moreImagesLabel: t("listings.morePhotos"),
          },
          fallbackUsed: false,
        }
      : {
          props: {
            iamgeUrl: fallbackImg,
            description: "",
          },
          fallbackUsed: true,
        }
  }
}

export const ListingDetailsImageCard = ({ listing }: ListingDetailsImageCardProps) => {
  const { getAssetPath } = useContext(ConfigContext)
  const listingAddress = getListingAddressString(listing)
  const shareButton = getAssetPath("share-button.svg")
  const shareButtonSelected = getAssetPath("share-button-selected.svg")
  const [shareImage, setShareImage] = useState(shareButton)

  const { fallbackUsed, props: imageCardProps } = createImageCardProps(listing)

  return (
    <header className="image-card--leader">
      <span aria-hidden={fallbackUsed}>
        <ImageCard
          {...imageCardProps}
          tags={
            listing.Reserved_community_type
              ? [{ text: getReservedCommunityType(listing.Reserved_community_type) }]
              : undefined
          }
        />
      </span>
      <div className="flex flex-col md:items-start md:text-left p-3 text-center">
        <h1 className="font-sans font-semibold text-2xl">{listing.Name}</h1>
        <p className="my-1">
          <ListingAddress listing={listing} />
        </p>
        <div className="flex flex-col items-center md:flex-row md:justify-between md:w-full">
          <div>
            <p className="text-gray-700">{listing.Developer}</p>
            <p className="my-2">
              <a
                href={`https://www.google.com/maps/place/${encodeURIComponent(listingAddress)}`}
                target="_blank"
                aria-label={t("label.opensInNewWindow", { linkText: t("label.viewOnMap") })}
              >
                {t("label.viewOnMap")}
              </a>
            </p>
          </div>
          <a
            href={`${getShareListingPath()}/${listing.listingID}`}
            onBlur={() => setShareImage(shareButton)}
            onFocus={() => setShareImage(shareButtonSelected)}
            onMouseEnter={() => setShareImage(shareButtonSelected)}
            onMouseLeave={() => setShareImage(shareButton)}
            aria-label={t("pageTitle.shareThisListing")}
            target="_blank"
            className="share-button"
          >
            <img alt={t("label.shareListView")} src={shareImage} />
          </a>
        </div>
      </div>
    </header>
  )
}
