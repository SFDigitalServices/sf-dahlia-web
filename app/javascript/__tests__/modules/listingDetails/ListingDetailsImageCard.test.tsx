import React from "react"
import { render } from "@testing-library/react"
import { ListingDetailsImageCard } from "../../../modules/listingDetails/ListingDetailsImageCard"
import { habitatListing } from "../../data/RailsSaleListing/listing-sale-habitat"
import { openSaleListing } from "../../data/RailsSaleListing/listing-sale-open"
import { saleEducatorListing } from "../../data/RailsSaleListing/listing-sale-educator"
import { rentalEducatorListing } from "../../data/RailsRentalListing/listing-rental-educator"

describe("ListingDetailsImageCard", () => {
  it("displays image card when no tag", () => {
    const { asFragment } = render(<ListingDetailsImageCard listing={openSaleListing} />)

    expect(asFragment()).toMatchSnapshot()
  })

  it("displays tag when listing is habitat for humanity", () => {
    const { asFragment } = render(<ListingDetailsImageCard listing={habitatListing} />)

    expect(asFragment()).toMatchSnapshot()
  })

  it("displays tag when sale listing is for educators", () => {
    const { asFragment } = render(<ListingDetailsImageCard listing={saleEducatorListing} />)

    expect(asFragment()).toMatchSnapshot()
  })

  it("displays tag when rental listing is for educators", () => {
    const { asFragment } = render(<ListingDetailsImageCard listing={rentalEducatorListing} />)

    expect(asFragment()).toMatchSnapshot()
  })
})
