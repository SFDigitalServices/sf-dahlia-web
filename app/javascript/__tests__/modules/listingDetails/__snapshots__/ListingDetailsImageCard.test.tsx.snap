// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingDetailsImageCard displays image card when no tag 1`] = `
<header
  className="image-card--leader"
>
  <a
    className="block"
    href="/listings/a0W0P00000GlKfBUAV"
  >
    <div
      className="image-card"
    >
      <div
        className="image-card-tag__wrapper"
      />
      <figure
        className="image-card__inner"
      >
        <img
          alt="A picture of the building"
          src="https://d31h23p2gsu1kr.cloudfront.net/images/listings/a0W0P00000GlKfBUAV-4d20089ed488fce1a695b73a8a4f090d.jpg"
        />
      </figure>
      <aside
        aria-label="A picture of the building Statuses"
        className="image-card__status"
      />
    </div>
  </a>
  <div
    className="flex flex-col md:items-start md:text-left p-3 text-center"
  >
    <h1
      className="font-sans font-semibold text-2xl"
    >
      TEST Sale Listing (do not modify) - Homeownership Acres
    </h1>
    <p
      className="my-1"
    >
      <span>
        1 South Van Ness Ave
        , 
        San Francisco
        ,
         
        <abbr
          className="no-underline"
          title="California"
        >
          CA
        </abbr>
         
        94103
      </span>
    </p>
    <div
      className="flex flex-col items-center md:flex-row md:justify-between md:w-full"
    >
      <div>
        <p
          className="text-gray-700"
        >
          Munoz, Carney and Clements
        </p>
        <p
          className="my-2"
        >
          <a
            aria-label="View on Map - Opens in a new window"
            href="https://www.google.com/maps/place/1%20South%20Van%20Ness%20Ave%2C%20San%20Francisco%2C%20CA%2094103"
            target="_blank"
          >
            View on Map
          </a>
        </p>
      </div>
      <a
        className="share-button"
        href="/share/a0W0P00000GlKfBUAV"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        target="_blank"
      >
        <img
          alt="Share"
        />
      </a>
    </div>
  </div>
</header>
`;

exports[`ListingDetailsImageCard displays tag when listing is habitat for humanity 1`] = `
<header
  className="image-card--leader"
>
  <a
    className="block"
    href="/listings/a0W8H000000HI2uUAG"
  >
    <div
      className="image-card"
    >
      <div
        className="image-card-tag__wrapper"
      >
        <span
          className="tag is-warning"
        >
          Habitat Greater San Francisco
        </span>
      </div>
      <figure
        className="image-card__inner"
      >
        <img
          alt="A picture of the building"
          src="https://d31h23p2gsu1kr.cloudfront.net/images/listings/a0W0P00000GlKfBUAV-4d20089ed488fce1a695b73a8a4f090d.jpg"
        />
      </figure>
      <aside
        aria-label="A picture of the building Statuses"
        className="image-card__status"
      />
    </div>
  </a>
  <div
    className="flex flex-col md:items-start md:text-left p-3 text-center"
  >
    <h1
      className="font-sans font-semibold text-2xl"
    >
      TEST: 36 Amber Drive
    </h1>
    <p
      className="my-1"
    >
      <span>
        36 Amber Drive
        , 
        San Francisco
        ,
         
        <abbr
          className="no-underline"
          title="California"
        >
          CA
        </abbr>
         
        94131
      </span>
    </p>
    <div
      className="flex flex-col items-center md:flex-row md:justify-between md:w-full"
    >
      <div>
        <p
          className="text-gray-700"
        />
        <p
          className="my-2"
        >
          <a
            aria-label="View on Map - Opens in a new window"
            href="https://www.google.com/maps/place/36%20Amber%20Drive%2C%20San%20Francisco%2C%20CA%2094131"
            target="_blank"
          >
            View on Map
          </a>
        </p>
      </div>
      <a
        className="share-button"
        href="/share/a0W8H000000HI2uUAG"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        target="_blank"
      >
        <img
          alt="Share"
        />
      </a>
    </div>
  </div>
</header>
`;
